src/docker/shared.rs

```diff
@@ -162,11 +162,11 @@ pub fn parse_docker_opts(value: &str) -> Result<Vec<String>> {
     shell_words::split(value).wrap_err_with(|| format!("could not parse docker opts of {}", value))
 }
 
-pub(crate) fn cargo_cmd(uses_xargo: bool) -> SafeCommand {
+pub(crate) fn cargo_safe_command(uses_xargo: bool) -> SafeCommand {
     if uses_xargo {
         SafeCommand::new("xargo")
     } else {
-        SafeCommand::new("cargo")
+        SafeCommand::new(env_program("CARGO", "cargo"))
```

Comment:
This is wrong, this struct should probably be renamed to `ContainedCommand`

It's used to assemble the command to use inside the container

we could introduce an environment variable `CROSS_CARGO_BIN` that sets this if wanted

Reply:
I think it's fine to skip that, since we need to mount a specific cargo right now.